proposals/channel-jamming-mitigation.md

```diff
@@ -0,0 +1,257 @@
+network, as they can leverage the jamming mitigation itself to further degrade 
+honest traffic.
+
+An upfront fee, paid regardless of the outcome of a payment attempt, is proposed
+to economically compensate nodes for providing traffic with access to liquidity
+and slots. Simulations in [2] show that a fee of as little as 1% of the success
+case fees are sufficient to compensate the opportunity cost of forwarding nodes
+in various routing scenarios.
+
+Nodes that advertise `option_upfront_fee` (56/57) will advertise a new TLV in
+their `channel_update` which expresses unconditional fees as a percentage of 
+their success-case fees advertised in the update: 
+```
+1. `tlv_stream`: `channel_update_tlvs`
+2. types:
+    1. type: 1 (`upfront_fee_policy`)
+    2. data:
+        * [`tu32`:`upfront_fee_base_ppm`]
+        * [`tu32`:`upfront_fee_proportional_ppm`]
+
+```
+
+To save the network the bandwidth and storage required to transmit and save
+defaults, nodes that advertise `option_upfront_fee` in their node announcement
+should be assumed to have a base and proportional fee that is 1% of its success
+case fees. Nodes also will not relay and senders will not utilize channels with
+a `channel_update` messages where upfront fees are > 10% of their success-case
+equivalent. This eliminates the risk of nodes setting these fees so high that
+it becomes economically rational for forwarding nodes to fail payments for the
+sake of their upfront fees. 
+
+The upfront fee amount should simply be assigned to the receiving party's 
+balance when an incoming HTLC is added. These amounts are not expected to be
+enforceable on-chain (as they are likely to be dust), so there is no need to 
+include an output for them in the channel's commitment transactions.
+
+As with success-case fees, upfront fees are charged on the outgoing link and 
+accumulated on the incoming link when payments are accepted for forwarding. The
+`update_add_htlc` message is extended with a new TLV which specifies the amount
+that must be pushed to the remote peer on addition of the incoming HTLC to its
+channel state. 
+```
+1. `tlv_stream`: `update_add_htlc_tlvs`
+2. types:
+    1. type: 2 (`upfront_fee_msat`)
+    2. data:
+        * [`tu64`:`upfront_fee_msat`]
+```
+
+The upfront fee that should be sent to the next peer in the route is provided
+in the onion's payload. As is currently the case with htlc forwarding amounts, 
+the difference between the incoming `upfront_fee_msat` amount in 
+`update_add_htlc_tlvs` and the outgoing `upfront_fee_to_forwad` is used to 
+validate that sufficient upfront fees are paid to the forwarding node. 
+```
+1. `tlv_stream`: `payload`
+1. type: 2 (`upfront_fee_to_forward`)
+    2. data:
+        * [`tu64`:`upfront_fee_to_forward`]
+```
+
+An upfront fee policy field is added to bolt-11 invoices to allow receiving 
+nodes to advertise the upfront fees that it will accept for the final hop 
```

Comment:
> to advertise the upfront fees that it will accept for the final hop

So IIUC: the receiver will also take an upfront fee, for the case where a failure occurs because e.g. they are not aware of the preimage that satisfies the proposed HTLC over that last hop

Is this introduced as a measure to prevent spam from direct peers that you have channels with?